pkg/controller/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -0,0 +1,200 @@
+package masteruserrecord
+
+import (
+	"context"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/kubefed/pkg/apis/core/common"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"testing"
+)
+
+const (
+	hostClusterName   = "host-cluster"
+	memberOperatorNs  = "toolchain-member-operator"
+	memberClusterName = "member-cluster"
+	hostOperatorNs    = "toolchain-host-operator"
+)
+
+type getMemberCluster func(cl client.Client) func(name string) (*cluster.FedCluster, bool)
+
+func TestCreateUserAccountSuccessful(t *testing.T) {
```

Comment:
What do you think about adding a test for the case when the UA already exists? Basically in this case nothing should fail or change (yet).

Reply:
When the UA already exists then it should synchronize and this is already covered in the following PR #23. This is why I skipped this case for this PR (to limit the conflicts and my work :-))

Reply:
Fair enough.